konrad/cloud.py

```diff
@@ -305,10 +370,52 @@ def __init__(self, z, cloud_fraction,
         self._interpolation_lw = self.interpolation_function(
             self._longwave_optical_thickness_per_meter, z)
 
+    def __add__(self, other):
+        """Define the superposition of two clouds in a layer."""
+        if not np.allclose(self.z, other.z):
+            raise ValueError('Incompatible `z` grids.')
+
+        cloud_fraction = np.max(
+            (self.cloud_area_fraction_in_atmosphere_layer,
+             other.cloud_area_fraction_in_atmosphere_layer),
+            axis=0,
+        ).clip(min=0, max=1)
+
+        lw_optical_thickness = np.sum(
+            (self.longwave_optical_thickness_due_to_cloud,
+             other.longwave_optical_thickness_due_to_cloud),
+            axis=0,
+        )
+        lw_optical_thickness = get_waveband_data_array(lw_optical_thickness,
+                                                       sw=False, numbands=16)
+
+        sw_optical_thickness = np.sum(
+            (self.shortwave_optical_thickness_due_to_cloud,
+             other.shortwave_optical_thickness_due_to_cloud),
+            axis=0,
+        )
+        sw_optical_thickness = get_waveband_data_array(sw_optical_thickness,
+                                                       sw=True, numbands=14)
+
+        summed_cloud = DirectInputCloud(
+            z=self.z,
+            cloud_fraction=get_p_data_array(cloud_fraction),
+            lw_optical_thickness=lw_optical_thickness,
+            sw_optical_thickness=sw_optical_thickness,
+            # TODO: Are they constant across numbands and mid_levels?
+            #  Should we use mean profiles/values/assume constants?
+            forward_scattering_fraction=0,
+            asymmetry_parameter=0.85,
+            single_scattering_albedo=0.9,
+        )
+
+        return summed_cloud
+
     def update_cloud_profile(self, atmosphere, **kwargs):
         """ Keep the cloud profile fixed with height.
         """
         z = atmosphere.get('z', keepdims=False)
+        self.z = z
         dz = dz_from_z(z)
```

Comment:
Do we need this?

Reply:
`self.z` is currently used to create the superposed cloud in the `CloudEnsemble`. I do not know if there is a better way. Does the cloud store the actual array for the `mid_levels`? I will check that, if yes, we could use these.

Reply:
Okay, fine. No, `mid_levels` just tells you what the coordinate is, but there are no values stored there.